lib/utils/reader.cpp

```diff
@@ -7,79 +7,101 @@ namespace utils{
   for(size_t i = 0; i < num_inequality_rows; ++i){
-    std::getline(newfile, temp_string);
-    matrix_row = convertStringToVector(temp_string);
+    std::getline(problems_filestream, temp_string);
+    problem_row = getProblemRowAsIntVector(temp_string);
+
+    constant_term = problem_row.at(0)*(-1);
+    problem_matrix_row = spliceVector(problem_row, 1, num_variables);
+
     //check vector size
-    if(matrix_row.size() != num_variables){
-      std::cout << "ERROR: length of matrix row is: " << matrix_row.size() << " but length of " << num_inequality_rows << " was expected" << std::endl;
+    if(problem_matrix_row.size() != num_variables){
+      std::cout << "ERROR: length of matrix row is: " << problem_row.size() << " but length of " << num_variables << " was expected" << std::endl;
     }
-    problem_matrix_.push_back(matrix_row);
-    matrix_row.clear();
+
+    problem_matrix_.push_back(problem_matrix_row);
+    upper_bounds_.push_back(kMaxInt);
+    lower_bounds_.push_back(constant_term);
+
+    problem_row.clear();
+    problem_matrix_row.clear();
   }
 
   //read in equality rows and add to problem matrix
-  std::string num_equality_rows_string;
-  std::getline(newfile, num_equality_rows_string);
-
-  core::int_t num_equality_rows = static_cast<core::int_t>(atoi(num_equality_rows_string.c_str()));
+  std::getline(problems_filestream, num_equality_rows_string);
+  int num_equality_rows = static_cast<int>(atoi(num_equality_rows_string.c_str()));
 
   for(size_t i = 0; i < num_equality_rows; ++i){
-    //get and typecast row vector
-    std::getline(newfile, temp_string);
-    matrix_row = convertStringToVector(temp_string);
+    std::getline(problems_filestream, temp_string);
+    problem_row = getProblemRowAsIntVector(temp_string);
+
+    constant_term = problem_row.at(0)*(-1);
+    problem_matrix_row = spliceVector(problem_row, 1, num_variables);
 
     //check vector size
-    if(matrix_row.size() != num_variables){
-      std::cout << "ERROR: Expexted row of length " << num_variables << " but row has length " << matrix_row.size() << std::endl;
+    if(problem_matrix_row.size() != num_variables){
+      std::cout << "ERROR: length of matrix row is: " << problem_row.size() << " but length of " << num_variables << " was expected" << std::endl;
     }
-    problem_matrix_.push_back(matrix_row);
-    matrix_row.clear();
-  }
-  newfile.close();
 
-  //make bound vectors
-  for(size_t i = 0; i < num_inequality_rows; ++i){
-    upper_bounds_.push_back(kMaxInt);
-    lower_bounds_.push_back(0);
-  }
+    problem_matrix_.push_back(problem_matrix_row);
+    upper_bounds_.push_back(constant_term);
+    lower_bounds_.push_back(constant_term);
 
-  for(size_t i = num_inequality_rows; i < num_equality_rows+num_inequality_rows; ++i){
-    upper_bounds_.push_back(0);
-    lower_bounds_.push_back(0);
+    problem_row.clear();
+    problem_matrix_row.clear();
   }
 
+  // Read the tilde line so that next problem starts
```

Comment:
multiline comments can be started with `/*` and ended a few lines later with `*/`